### ISSUE TITLE
Autograd will take `init` module API into account when using `jit`

### ISSUE THREAD

**Issue by payphon** [pytorch/pytorch#84673]
### 🐛 Describe the bug

As mentioned in doc,
> All the functions in this module (`init`) are intended to be used to initialize neural network parameters, so they all run in :func:`torch.no_grad` mode and will not be taken into account by autograd.

However, autograd will take `init` module API into account when using `jit`.

```py
import torch

def fn(input):
    return torch.nn.init.ones_(input)

input = torch.rand([4])
jit_fn = torch.jit.trace(fn, (input.clone(), ))

fn(input.clone().requires_grad_()) # work
jit_fn(input.clone().requires_grad_())
# RuntimeError: a leaf Variable that requires grad is being used in an in-place operation.
```

### Versions

pytorch: 1.12.1

**Comment by davidberard98** [comment]
I suggest wrapping such calls with `with torch.no_grad():` as a workaround.

**Comment by payphon** [comment]
Yes, `no_grad` does work. But it might be better to have a consistent behavior for JIT.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
